### PR TITLE
add an absolute measure of election score on-chain as a parameter

### PR DESCRIPTION
**Pull request by kianenigma**
I'm thinking of making it a storage item that can be easily updated via the governance, otherwise it is the same thing. 

polkadot companion: https://github.com/paritytech/polkadot/pull/3099

### CODE REVIEW

**Review by andresilva — COMMENTED** [review]
lgtm but I don't know anything about this code.

File: frame/election-provider-multi-phase/src/lib.rs [hunk 1]
```diff
@@ -773,6 +778,21 @@ pub mod pallet {
 
 			Ok(None.into())
 		}
+
+		/// Set a new value for `MinimumUntrustedScore`.
+		///
+		/// Dispatch origin must be aligned with `T::ForceOrigin`.
+		///
+		/// This check can be turned off by setting the value to `None`.
+		#[pallet::weight(T::DbWeight::get().reads(1))]
```
Comment (shawntabrizi) [inline frame/election-provider-multi-phase/src/lib.rs:787]:
```suggestion
		#[pallet::weight(T::DbWeight::get().reads_writes(1, 1))]
```

File: frame/election-provider-multi-phase/src/lib.rs [hunk 2]
```diff
@@ -773,6 +778,21 @@ pub mod pallet {
...
+		fn set_minimum_untrusted_score(
+			origin: OriginFor<T>,
+			maybe_next_score: Option<ElectionScore>,
+		) -> DispatchResultWithPostInfo {
+			T::ForceOrigin::ensure_origin(origin)?;
+			<MinimumUntrustedScore<T>>::set(maybe_next_score);
+			Ok(None.into())
```
Comment (shawntabrizi) [inline frame/election-provider-multi-phase/src/lib.rs:794]:
```suggestion
			Ok(())
```

Why is this none?

File: frame/election-provider-multi-phase/src/lib.rs [hunk 3]
```diff
@@ -773,6 +778,21 @@ pub mod pallet {
...
+		/// Set a new value for `MinimumUntrustedScore`.
+		///
+		/// Dispatch origin must be aligned with `T::ForceOrigin`.
+		///
+		/// This check can be turned off by setting the value to `None`.
+		#[pallet::weight(T::DbWeight::get().reads(1))]
+		fn set_minimum_untrusted_score(
+			origin: OriginFor<T>,
+			maybe_next_score: Option<ElectionScore>,
+		) -> DispatchResultWithPostInfo {
```
Comment (shawntabrizi) [inline frame/election-provider-multi-phase/src/lib.rs:791]:
```suggestion
		) -> DispatchResult {
```

File: frame/election-provider-multi-phase/src/lib.rs (hunk 1 above)
Comment (thiolliere) [inline frame/election-provider-multi-phase/src/lib.rs:787]:
```suggestion
		#[pallet::weight(T::DbWeight::get().writes(1))]
```

**Review by thiolliere — APPROVED** [review]
looks good to me

**Review by shawntabrizi — APPROVED** [review]

**Review by coriolinus — APPROVED** [review]
Nothing to add that hasn't already been addressed.

File: frame/election-provider-multi-phase/src/lib.rs [hunk 4]
```diff
@@ -1052,6 +1080,15 @@ impl<T: Config> Pallet<T> {
 		// upon arrival, thus we would then remove it here. Given overlay it is cheap anyhow
 		ensure!(winners.len() as u32 == desired_targets, FeasibilityError::WrongWinnerCount);
 
+		// ensure that the solution's score can pass absolute min-score.
+		let submitted_score = solution.score.clone();
+		ensure!(
+			Self::minimum_untrusted_score().map_or(true, |min_score|
```
Comment (KiChjang) [inline frame/election-provider-multi-phase/src/lib.rs:1086]: By having this check here, would it affect the weight of `submit_unsigned`? Or is this operation minuscule enough to not have any noticeable effect?

Comment (kianenigma) [reply frame/election-provider-multi-phase/src/lib.rs:1086]: insignificant to reweigh.

**Comment by kianenigma** [comment]
bot merge

**Comment by kianenigma** [comment]
bot merge damn it.

**Comment by kianenigma** [comment]
bot merge

**Pull request merged**